### PR TITLE
Expand MiningStatistic component functionality: Introduce BoxDisplayer

### LINKED ISSUE #448: Expand MiningStatistic component functionality
In Chapter 3 we'll want to use the same `MiningStatistic` component that @vinayaksh42 created in  #349 but expand its functionality so it supports the variants we need.

### Design
image

Variants:
image
image
image
image

### PR DESCRIPTION
**Pull request by shaavan**
This PR, along with #535

resolves #448

Introduce the BoxDisplay component to display the percentage for the user.

**Screenshot:**

|Green Background|Transparent Background|
|---|---|
|Screenshot 2023-06-30 at 7 09 47 PM|Screenshot 2023-06-30 at 7 10 00 PM|

### CODE REVIEW

File: content/lessons/chapter-2/mining-1.tsx [hunk 1]
```diff
@@ -295,6 +296,17 @@ export default function Mining1({ lang }) {
...
+            <TitleCard title={'Hash Rate'} disabled={false}></TitleCard>
+            <BoxDisplayer className="ml-10">
+              <span>25.00%</span>
+            </BoxDisplayer>
+            <HashFrequency
+              disabled={nonce === 0}
+              step={step}
+              hashPower={hashPower}
+            />
+          </Card>
```
Comment (benalleng) [inline content/lessons/chapter-2/mining-1.tsx:309]:
``` suggestion
```

File: content/lessons/chapter-2/mining-1.tsx [hunk 2]
```diff
@@ -11,6 +11,7 @@ import {
   TitleCard,
   HashFrequency,
   StartButton,
+  BoxDisplayer,
```
Comment (benalleng) [inline content/lessons/chapter-2/mining-1.tsx:14]:
``` suggestion
```

**Review by benalleng — APPROVED** [review]
LGTM! lets just remove the test in mining-1 and then we should be good to merge!

**Comment by shaavan** [comment]
Updated from 8eadabd2dc3380a6befabdef54228331f03209b5 to 8852c948c97a12026a90871b9b7c7208e8ebd52b ([pr540.01](https://github.com/shaavan/saving-satoshi/commits/pr540.01) -> [pr540.02](https://github.com/shaavan/saving-satoshi/commits/pr540.02), [diff](https://github.com/shaavan/saving-satoshi/compare/pr540.01..pr540.02))
Addressed @benalleng comment

- Removed test commit.

File: content/lessons/chapter-2/mining-1.tsx (hunk 1 above)
Comment (shaavan) [reply content/lessons/chapter-2/mining-1.tsx]: Addressed in 8852c948c97a12026a90871b9b7c7208e8ebd52b
Thanks!

File: content/lessons/chapter-2/mining-1.tsx (hunk 2 above)
Comment (shaavan) [reply content/lessons/chapter-2/mining-1.tsx]: Addressed in 8852c948c97a12026a90871b9b7c7208e8ebd52b
Thanks for reviewing!

**Pull request merged**